packages/webpack-cli/lib/utils/Compiler.js

```diff
@@ -18,24 +18,24 @@ class Compiler {
 
         compilation.hooks.beforeRun.tap('webpackProgress', () => {
             if (outputOptions.progress) {
-                process.stdout.write('\n');
+                logger.raw('\n');
                 const defaultProgressPluginHandler = (percent, msg) => {
                     percent = Math.floor(percent * 100);
                     readline.clearLine(process.stdout, 0);
                     readline.cursorTo(process.stdout, 0, null);
                     if (percent !== undefined) {
-                        process.stdout.write(' (');
+                        logger.raw(' (');
                         for (let i = 0; i <= 100; i += 10) {
                             if (i <= percent) {
-                                process.stdout.write(greenBright('#'));
+                                logger.raw(greenBright('#'));
                             } else {
-                                process.stdout.write('#');
+                                logger.raw('#');
                             }
                         }
-                        process.stdout.write(`) ${percent}% : `);
-                        process.stdout.write(`${cyanBright(msg)}`);
+                        logger.raw(`) ${percent}% : `);
+                        logger.raw(`${cyanBright(msg)}`);
                         if (percent === 100) {
-                            process.stdout.write(`${cyanBright('Compilation completed\n')}`);
+                            logger.raw(`${cyanBright('Compilation completed\n')}`);
```

Comment:
We should remove this stuff, because it is out of scope webpack-cli, but let's do it in separate PR, I am afraid it also create a lot of unnecessary newlines

Reply:
Related to whole code in this file